batch/batch/front_end/front_end.py

```diff
@@ -683,13 +683,22 @@ async def create_jobs(request, userdata):
                     if user != 'ci':
                         raise web.HTTPBadRequest(reason=f'unauthorized secret {(secret["namespace"], secret["name"])}')
 
+                env = spec.get('env')
+                if not env:
+                    env = []
+                    spec['env'] = env
+
                 spec['secrets'] = secrets
                 secrets.append({
                     'namespace': DEFAULT_NAMESPACE,
                     'name': userdata['gsa_key_secret_name'],
                     'mount_path': '/gsa-key',
                     'mount_in_copy': True
                 })
+                if all('GOOGLE_APPLICATION_CREDENTIALS' != envvar['name'] for envvar in spec.env):
```

Comment:
Why did you break up the env definition with the secrets?

Reply:
I guess I thought of this block as part of the secrets and moving the env down here would break up the secrets definition. I'll switch.